### PR TITLE
Restructure the Cannot Connect to a Service troubleshooting guide

### LINKED ISSUE #13306: Restructure the Cannot connect to a service exposed by an APIRule guide
**Description**

The [**Cannot connect to a service exposed by an APIRule** ](https://kyma-project.io/docs/kyma/latest/04-operation-guides/troubleshooting/apix-01-apigateway-connect-api-rule/) troubleshooting guide is long and does not follow the [template](https://github.com/kyma-project/community/blob/main/templates/resources/troubleshooting.md). Restructure the document and preferably divide it into two parts for better readability.

**Assignees**

@kyma-project/technical-writers

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes proposed in this pull request:

- Restructured the **Cannot Connect to a Service...** troubleshooting guide

**Related issue(s)**
Resolves #13306

### CODE REVIEW

**Review by strekm — DISMISSED** [review]
/meow

File: docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+- Proper VirtualService has been created:
+
+  ```bash
+  kubectl get virtualservices.networking.istio.io -n {NAMESPACE}
+  ```
+
+  >**TIP:** Name of the VirtualService consists of the name of the API Rule and a random suffix.
+
+Sometimes Oathkeeper Maester controller stops reconciling Rules on long-living clusters. This can result in random `404 Not Found` responses, because Oathkeeper does not contain Rules reflecting the actual state of the cluster. A simple restart of the Pod resolves the issue, but you might want to verify if that is the issue you have encountered. To do so:
```
Comment (majakurcius) [inline docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md:29]:
```suggestion
Sometimes Oathkeeper Maester controller stops reconciling Rules on long-living clusters. This can result in random `404 Not Found` responses because Oathkeeper does not contain Rules reflecting the actual state of the cluster. A simple restart of the Pod resolves the issue, but you might want to verify if that is the issue you have encountered. To do so, follow these steps:
```

File: docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+3. If you have more than one instance of Oathkeeper, compare whether the files contain the same Rules. Oathkeeper stores Rules as JSON files, so you might want to use [jd](https://github.com/josephburnett/jd) to automate the comparison:
+
+    ```bash
+   jd -set {FIRST_FILE} {SECOND_FILE} 
+   ```
+
+    Oathkeeper Pods are out of sync if there are any differences between the files other than the order of Rules.
+   
+4. Compare the Rules in the files with Rules present on the cluster. If there are any differences, Oathkeeper Pods are out of sync.
```
Comment (majakurcius) [inline docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md:51]: These sentences are kind of a duplication. I get that in the first one you want to explain what actually counts as a "difference". How about rewording the sentences as such? 
```suggestion
    The files are considered different by jd if there are any differences between the files other than the order of Rules.
   
4. Compare the Rules in the files with Rules present on the cluster. If the files are different, Oathkeeper Pods are out of sync.
```

Or maybe you have a better idea?

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      ```
+
+  3. Check logs from the `ory-curl` Pod:
+
+      ```bash
+      kubectl logs -n {SECRET_NAMESPACE} ory-curl curl
+      ```
+
+  4. If the Client ID from step 1 is not available on the clients list, make sure Hydra has access to the database and/or restart the Hydra Measter Pod.
+  You can check the logs using the following commands:
```
Comment (majakurcius) [inline docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md:75]: Be aware that these are rendered as one line (and the numbering uses Roman numbers, as it's under an unordered list item). 
image

File: docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  kubectl get rules.oathkeeper.ory.sh -n {NAMESPACE}
+  ```
+
+  >**TIP:** Name of the Rule consists of the name of the API Rule and a random suffix.
+
+- Proper VirtualService has been created:
+
+  ```bash
+  kubectl get virtualservices.networking.istio.io -n {NAMESPACE}
+  ```
```
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md:25]: Line 11 says:

> Make sure that the following conditions are met:
 So I think it's clear that the commands check the status.

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md (hunk 3 above)
Comment (mmitoraj) [reply docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md]: I fixed the formatting so that the commands are in two separate code snippets.

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+title: Cannot connect to a service exposed by an API Rule - 401 Unauthorized or 403 Forbidden
+---
+
+## Symptom
+
+You reach your service and get `401 Unauthorized` or `403 Forbidden` in response.
+
+## Remedy
+
+In such case, make sure that:
```
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md]: Actually, just FYI, I checked 'cause I wasn't sure, and apparently "in such case" is also correct. Touche :)

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+- You are using an access token with proper scopes and it is active:
+
+  1. Export the credentials of your OAuth2Client as environment variables:
+
+      ```bash
+      export CLIENT_ID="$(kubectl get secret -n $CLIENT_NAMESPACE $CLIENT_NAME -o jsonpath='{.data.client_id}' | base64 --decode)"
+      export CLIENT_SECRET="$(kubectl get secret -n $CLIENT_NAMESPACE $CLIENT_NAME -o jsonpath='{.data.client_secret}' | base64 --decode)"
```
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md:19]: In that case, I'd either write this explicitly (maybe in a note), or replace it with a standard placeholder `{CLIENT_NAMESPACE}`, `{CLIENT_SECRET}`.

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  2. Encode your client credentials and export them as an environment variable:
+
+      ```bash
+      export ENCODED_CREDENTIALS=$(echo -n "$CLIENT_ID:$CLIENT_SECRET" | base64)
+      ```
+
+  3. Check access token status:
+
+      ```bash
+      curl -X POST "https://oauth2.{DOMAIN}/oauth2/introspect" -H "Authorization: Basic $ENCODED_CREDENTIALS" -F "token={ACCESS_TOKEN}"
```
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md:31]: Sorry, should have explained. 
For consistency's sake. `CLUSTER_DOMAIN` is used in Get Started guides and Application Connector documentation, and it's more specific than just `DOMAIN`. I thought it would be good to use the same placeholder throughout the whole documentation.

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+     ```bash
+      # Check logs from the Hydra-Maester controller application
+      kubectl logs -n kyma-system -l "app.kubernetes.io/name=hydra-maester" -c hydra-maester
+      # Example output
+      2020-05-04T12:19:04.472Z  INFO  controller-runtime.controller Starting EventSource  {"controller": "oauth2client", "source": "kind source: /, Kind="}2020-05-04T12:19:04.472Z  INFO  setup starting manager
+      2020-05-04T12:19:04.573Z  INFO  controller-runtime.controller Starting Controller {"controller": "oauth2client"}
+      2020-05-04T12:19:04.673Z  INFO  controller-runtime.controller Starting workers  {"controller": "oauth2client", "worker count": 1}
+      2020-05-04T12:26:30.819Z  INFO  controllers.OAuth2Client  using default client
+      2020-05-04T12:26:30.835Z  INFO  controllers.OAuth2Client  using default client
+      # This log entry informs that a client has been created, and should be visible within hydra
```
Comment (majakurcius) [inline docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md:86]:
```suggestion
      # This log entry informs us that a client has been created, and should be visible within Hydra
```
Sorry, I missed this one before. This is minor, but `inform` is a transitive verb and needs an object (`us`/`you`). See the [Cambridge Dictionary](https://dictionary.cambridge.org/dictionary/english/inform) for reference.

File: docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/04-operation-guides/troubleshooting/apix-03-404-not-found.md:25]: OK

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md (hunk 3 above)
Comment (majakurcius) [reply docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md]: Oh? I still see it [displayed as one line](https://github.com/kyma-project/kyma/blob/b8d0bd064e60103925f98c4cdb8b49737d9820bc/docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md). Not that I mind, it's perfectly fine, I just noticed that you put the second sentence in a new line, so wanted to make sure you're aware of it. 
image.

I have an impression we're talking about two different things 🤔

File: docs/01-overview/main-areas/api-exposure/apix-02-useful-links.md [hunk 9]
```diff
@@ -9,15 +9,16 @@ If you're interested in learning more about API Exposure in Kyma, follow these l
...
-  - [Connection refused](../../../04-operation-guides/troubleshooting/apix-02-dns-mgt-connection-refused.md) errors occur when you want to use your custom domain
-  - You receive the [`could not resolve host`](../../../04-operation-guides/troubleshooting/apix-03-dns-mgt-could-not-resolve-host.md) error when you want to use your custom domain
-  - A [resource is ignored by the controller](../../../04-operation-guides/troubleshooting/apix-04-dns-mgt-resource-ignored.md)
-  - The [Issuer Custom Resource fails to be created](../../../04-operation-guides/troubleshooting/apix-05-cert-mgt-issuer-not-created.md)
-  - The [Kyma Gateway is not reachable](../../../04-operation-guides/troubleshooting/apix-06-gateway-not-reachable.md)
+  - You get the [401 Unauthorized or 403 Forbidden](../../../04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md) status code when you try to connect to a service exposed by an API Rule
+  - You get the [404 Not Found](../../../04-operation-guides/troubleshooting/apix-03-404-not-found.md) status code when you try to connect to a service exposed by an API Rule
```
Comment (majakurcius) [inline docs/01-overview/main-areas/api-exposure/apix-02-useful-links.md:16]:
```suggestion
  - You get the [`401 Unauthorized` or `403 Forbidden`](../../../04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md) status code when you try to connect to a service exposed by an API Rule
  - You get the [`404 Not Found`](../../../04-operation-guides/troubleshooting/apix-03-404-not-found.md) status code when you try to connect to a service exposed by an API Rule
```
I also remembered we had these [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#use-code-font-for-these-items) in place to put status codes in code font.

File: docs/04-operation-guides/troubleshooting/apix-01-apigateway-connect-api-rule.md [hunk 10]
```diff
@@ -1,142 +1,14 @@
...
-    ```bash
-   jd -set {FIRST_FILE} {SECOND_FILE} 
-   ```
+If the status is `Error`, edit the API Rule and fix issues described in `.Status.APIRuleStatus.Desc`. If you still encounter issues, make sure the API Gateway, Hydra, and Oathkeeper are running or take a look at one of the more specific troubleshooting guides:
 
-    Oathkeeper Pods are out of sync if there are any differences between the files other than the order of Rules.
-   
-4. Compare the Rules in the files with Rules present on the cluster. If there are any differences, Oathkeeper Pods are out of sync.
+- [Cannot connect to a service exposed by an API Rule - 401 Unauthorized or 403 Forbidden](./apix-02-401-unauthorized-403-forbidden.md)
+- [Cannot connect to a service exposed by an API Rule - 404 Not Found](./apix-03-404-not-found.md)
```
Comment (majakurcius) [inline docs/04-operation-guides/troubleshooting/apix-01-apigateway-connect-api-rule.md:14]:
```suggestion
- [Cannot connect to a service exposed by an API Rule - `401 Unauthorized` or `403 Forbidden`](./apix-02-401-unauthorized-403-forbidden.md)
- [Cannot connect to a service exposed by an API Rule - `404 Not Found`](./apix-03-404-not-found.md)
```

File: docs/01-overview/main-areas/api-exposure/apix-02-useful-links.md (hunk 9 above)
Comment (mmitoraj) [reply docs/01-overview/main-areas/api-exposure/apix-02-useful-links.md]: Yes, we do. I used code font in the troubleshooting docs but forgot here.

File: docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/04-operation-guides/troubleshooting/apix-02-401-unauthorized-403-forbidden.md:19]: I added a note.

**Review by majakurcius — APPROVED** [review]
🚀
